### ISSUE TITLE
detecting/filling wrong fields

### ISSUE THREAD

**Issue by Znrl** [keepassxreboot/keepassxc-browser#2022]
## Expected Behavior
Keepass/the browser Addon should not detect those fields as Login fields. Or make it possible to exclude them.

Well, I tried to search... but I couldn't really find anything.
Is there already a way to customize a blacklist or similar?
Create some rules...?

## Current Behavior

https://www.sparkasse-staufen-breisach.de

It happens with activated auto-fill and the option for username-only option.

This also results in a "jump" down there on loading the site.

Keepass keeps detecting the search field in the section "Börsen und Märkte" close to the bottom of the page as login field

`<input name="such" type="text" placeholder="WKN, ISIN, Bezeichnung">`
I also notice, that the "web.s-investor.de" is added alongside the actual URL to the properties in Keepass {"Allow"; []...
Just a side note, but I would think, it would be nice to be able to just allow the main URL?

One problem here might be, that they use quite cryptic names and IDs for their fields.

This also lead to keepass autofilling a "date" field when trying to make a transfer, it keeps filling it with the Login Details and is quite a hassle.
The date field looks like this:
`<input type="text" name="xLyaSvsLurehcnMX" id="xLyaSvsLurehcnMX" value="" class="nbf-dateinput" maxlength="10" placeholder="31.10.2023" data-name="openDatePicker">
`

I know that this is not the only bank that uses basically the same system, see "bw-bank.de".
But they seem to not have some other input field on their site, so it works.

## Possible Solution

Idk how to do this automatically.
But what I would really like to see is:

Right click the field, choose "KeePass-XC-Browser" and then have a option "ignore field".
Basically a blacklist.

But I know that this might not be that easy, since name and ID of the field are probably newly generated every time.
It would need to recognize it from context, class names, ...

One thing that somehow works it to use custom fields. But it seems to break the auto fill.

## Steps to Reproduce (for bugs)

1. create some credentials for https://www.sparkasse-staufen-breisach.de, enable auto fill
2. visit https://www.sparkasse-staufen-breisach.de
3. click "Anmelden" (top right)
4. activate the username-only option
5. revisit https://www.sparkasse-staufen-breisach.de
6. it should now jump to the bottom and fill in the search field

## Debug info

KeePassXC - {2.7.6}
KeePassXC-Browser - {1.8.9}
Operating system: Win
Browser: Firefox

**Comment by varjolintu** [comment]
I did a test entry with `https://sparkasse-staufen-breisach.de/de/home.html` as the URL, and added the Username-Only option from the extension popup as it suggested it. Everything works fine.

About the search field, there's not much to be done. The page doesn't really identify the field as a search field, so it's not ignored by the extension. If you don't use Auto-Submit or Auto-Fill, this shouldn't be a problem.

**Comment by Znrl** [comment]
I get it... In my eyes it's badly done on the site.
It kind of identifies it. But it uses German terms for id/name tag "suchid" and "such" (imperative form of "suchen" which would be translated to "searchid" or "search".

My problem is, that I'm not the user in this case, and for for a lot of "normal" users it just needs to work with auto fill and no buttons. :/

**Comment by varjolintu** [comment]
Even if the icons are sometimes not visible in the login input field, the Autocomplete Menu still works when you click at the input field.
But there's nothing to do with the search field, unfortunately.
